### PR TITLE
SNOW-1361785 Run e2e jar tests with multiple Java versions

### PR DESCRIPTION
**Pull request by sfc-gh-lsembera**
This PR runs e2e jar tests with all LTS Java versions on all supported clouds. Compilation always happens with Java 8. The e2e jar test has been enhanced to test all supported data types. Additionally, this PR integrates google format with the e2e jar test project, so some changes lines are just reformats.

### CODE REVIEW

File: format.sh [hunk 1]
```diff
@@ -17,8 +18,12 @@ if ! command -v java > /dev/null; then
   echo "Java not installed."
   exit 1
 fi
-echo "Running Google Java Format"
-find ./src -type f -name "*.java" -print0 | xargs -0 java -jar "${JAR_FILE}" --replace --set-exit-if-changed && echo "OK"
 
-echo "Sorting pom.xml"
```
Comment (sfc-gh-tzhang) [inline format.sh:23]: nit: remove by accident?

**Review by sfc-gh-tzhang — APPROVED** [review]
LGTM, thanks!

**Comment by sfc-gh-tzhang** [comment]
> Compilation always happens with Java 8. The e2e jar test has been enhanced to test all supported data types.

Do we want to verify compilation with other java versions as well?

**Comment by sfc-gh-lsembera** [comment]
> Do we want to verify compilation with other java versions as well?

@sfc-gh-tzhang I tried it within a timebox, but didn't get it working. It is not necessary, though, because customers will anyway use the SDK from Maven, which we always publish with Java 8.

File: format.sh (hunk 1 above)
Comment (sfc-gh-lsembera) [reply format.sh:23]: fixed

**Pull request merged**